code/modules/mining/mine_turfs.dm

```diff
@@ -70,6 +70,24 @@ var/list/mining_overlay_cache = list()
 	clear_ore_effects()
 	. = ..()
 
+/turf/simulated/mineral/xenoarch
+	name = "random digsite"
+
+/turf/simulated/mineral/xenoarch/Initialize()
+	. = ..()
+	name = "rock"
+
+	make_digsite(rand(0,2))
```

Comment:
`prob(50)`?

Reply:
make_digsite's first input is a range, not probability.